packages/observe-sequence/observe_sequence_tests.js

```diff
@@ -205,6 +205,19 @@ Tinytest.add('observe-sequence - initial data for all sequence types', function
     {addedAt: ["13", {_id: "13", foo: 1}, 0, null]},
     {addedAt: [{NOT: "13"}, {_id: "13", foo: 2}, 1, null]}
   ], /*numExpectedWarnings = */1);
+
+  // non-array iterable (empty)
+  if(typeof Map == 'function') runOneObserveSequenceTestCase(test, function () {
+    return new Map();
+  }, function () {}, []);
+
+  // non-array iterable (non-empty)
+  if(typeof Set == 'function') runOneObserveSequenceTestCase(test, function () {
+    return new Set([{foo: 1}, {bar: 2}]);
+  }, function () {}, [
+    {addedAt: [0, {foo: 1}, 0, null]},
+    {addedAt: [1, {bar: 2}, 1, null]}
+  ]);
```

Comment:
Should we add a few more tests here? I think if we publish this and productions will use this there may be unforseen edge-cases.

I would propose to add tests for:

- [x] empty map
- [ ] arbitrary filled map
- [ ] empty set
- [x] arbitrary filled set
- [ ] empty custom structure implementing the iterable protocol
- [ ] arbitrary filled custom structure implementing the iterable protocol

Reply:
More tests are always welcome.